RuntimeException if no application selected to choose a file to import accounts

GnuCash crashes with a `RuntimeException` if no application is selected from the dialog that appears when importing transactions.

Steps to reproduce:

1.  Select _Settings > Import GnuCash Accounts_.
2. A dialog appears with different applications installed in the device that can be used to choose a file to import.
3. Press Back button (from old Android device). Any other action that closes the activity seems to also trigger the issue.
4. The application crashes with the following stack trace:
```
org.gnucash.android.devel E/AndroidRuntime﹕ FATAL EXCEPTION: main
    java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=1, result=0, data=null} to activity {org.gnucash.android.devel/org.gnucash.android.ui.settings.SettingsActivity}: java.lang.NullPointerException
            at android.app.ActivityThread.deliverResults(ActivityThread.java:2532)
            at android.app.ActivityThread.handleSendResult(ActivityThread.java:2574)
            at android.app.ActivityThread.access$2000(ActivityThread.java:117)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:961)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:130)
            at android.app.ActivityThread.main(ActivityThread.java:3683)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:507)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:878)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:636)
            at dalvik.system.NativeStart.main(Native Method)
     Caused by: java.lang.NullPointerException
            at org.gnucash.android.ui.settings.SettingsActivity.onActivityResult(SettingsActivity.java:577)
            at android.app.Activity.dispatchActivityResult(Activity.java:3914)
            at android.app.ActivityThread.deliverResults(ActivityThread.java:2528)
            at android.app.ActivityThread.handleSendResult(ActivityThread.java:2574)
            at android.app.ActivityThread.access$2000(ActivityThread.java:117)
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:961)
            at android.os.Handler.dispatchMessage(Handler.java:99)
            at android.os.Looper.loop(Looper.java:130)
            at android.app.ActivityThread.main(ActivityThread.java:3683)
            at java.lang.reflect.Method.invokeNative(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:507)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:878)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:636)
            at dalvik.system.NativeStart.main(Native Method)
```

--- a/app/src/main/java/org/gnucash/android/ui/settings/SettingsActivity.java
+++ b/app/src/main/java/org/gnucash/android/ui/settings/SettingsActivity.java
@@ -573,12 +573,14 @@ public class SettingsActivity extends SherlockPreferenceActivity implements OnPr
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         switch (requestCode) {
             case AccountsActivity.REQUEST_PICK_ACCOUNTS_FILE:
-                try {
-                    InputStream accountInputStream = getContentResolver().openInputStream(data.getData());
-                    new ImportAsyncTask(this).execute(accountInputStream);
-                } catch (FileNotFoundException e) {
-                    Crashlytics.logException(e);
-                    Toast.makeText(this, R.string.toast_error_importing_accounts, Toast.LENGTH_SHORT).show();
+                if (resultCode == Activity.RESULT_OK && data != null) {
+                    try {
+                        InputStream accountInputStream = getContentResolver().openInputStream(data.getData());
+                        new ImportAsyncTask(this).execute(accountInputStream);
+                    } catch (FileNotFoundException e) {
+                        Crashlytics.logException(e);
+                        Toast.makeText(this, R.string.toast_error_importing_accounts, Toast.LENGTH_SHORT).show();
+                    }
                 }
                 break;
             case PasscodePreferenceFragment.PASSCODE_REQUEST_CODE:

